mempool/mempool.go

```diff
@@ -0,0 +1,205 @@
+package mempool
+
+import (
+	"context"
+	"database/sql"
+	"math"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v2"
+	dcrjson "github.com/decred/dcrd/rpc/jsonrpc/types/v2"
+	"github.com/decred/dcrd/wire"
+	"github.com/planetdecred/pdanalytics/dcrd"
+	"github.com/planetdecred/pdanalytics/web"
+)
+
+func NewCollector(ctx context.Context, client *dcrd.Dcrd, interval float64,
+	dataStore DataStore, webServer *web.Server) (*Collector, error) {
+
+	c := &Collector{
+		ctx:                ctx,
+		webServer:          webServer,
+		client:             client,
+		collectionInterval: interval,
+		dataStore:          dataStore,
+	}
+
+	if err := c.SetExplorerBestBlock(ctx); err != nil {
+		return nil, err
+	}
+
+	hash, err := client.Rpc.GetBestBlockHash()
+	if err != nil {
+		return nil, err
+	}
+	blockHeader, err := client.Rpc.GetBlockHeader(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = c.ConnectBlock(blockHeader); err != nil {
+		return nil, err
+	}
+
+	client.Notif.RegisterBlockHandlerGroup(c.ConnectBlock)
+
+	if err := c.webServer.Templates.AddTemplate("mempool"); err != nil {
+		log.Errorf("Unable to create new html template: %v", err)
+		return nil, err
+	}
+
+	c.webServer.AddMenuItem(web.MenuItem{
+		Href:      "/mempool",
+		HyperText: "Mempool",
+		Attributes: map[string]string{
+			"class": "menu-item",
+			"title": "Historic mempool data",
+		},
+	})
+
+	webServer.AddRoute("/mempool", web.GET, c.mempoolPage)
+	webServer.AddRoute("/getmempool", web.GET, c.getMempool)
+	webServer.AddRoute("/api/charts/mempool/{chartDataType}", web.GET, c.chart, chartDataTypeCtx)
+
+	return c, nil
+}
+
+func (c *Collector) SetExplorerBestBlock(ctx context.Context) error {
+	var explorerUrl string
+	switch c.client.Params.Name {
+	case chaincfg.MainNetParams().Name:
+		explorerUrl = "https://explorer.dcrdata.org/api/block/best" //TODO: use dcrd server
```

Comment:
Yes this function should use the dcrd client.

Reply:
I am having an issue with using the rpc client.
This function is used to determine the sync status of the node
`if the incoming block is less than the best block, then the client is not fully synced yet and so we wont take mempool snapshot`
If we use the `getbestblock` JSON-RPC command, it will return the height of the block in the longest (best) chain of the client which will not be the actual best block because the client is not fully synced yet.

Reply:
There is no need to check sync status 
Snapshot can be taken while dcrd is syncing